packages/core/src/modules/discover-features/protocol/v2/V2DiscoverFeaturesService.ts

```diff
@@ -83,18 +88,25 @@ export class V2DiscoverFeaturesService extends DiscoverFeaturesService {
 
   public async createDisclosure(
     options: CreateDisclosureOptions
-  ): Promise<DiscoverFeaturesProtocolMsgReturnType<V2DisclosuresMessage>> {
+  ): Promise<DiscoverFeaturesProtocolMsgReturnType<V2DisclosuresMessage | V2DisclosuresDidCommV2Message>> {
     const matches = this.featureRegistry.query(...options.disclosureQueries)
 
-    const discloseMessage = new V2DisclosuresMessage({
-      threadId: options.threadId,
-      features: matches,
-    })
+    const discloseMessage = options.connectionRecord?.isDidCommV2Connection
```

Comment:
To extend on my proposal to infer which didomm version to use based on the services. Maybe there can be a `toDidCommV1Message` on the v2 base class and a `toDidCommV2Message` on the v1 base class and that this is optionally implemented when you support 2 didcomm versions for hte same protocol version. Then we could automatically transform it into the other type when sending the message if the method is available, and the didcomm version does not match.

Reply:
But it doens't make fully sense, because we do need to know upfront what didcomm version a connection supports. And we should be able to update that in the connection record when we re-fetch a did and resolve their servides (so it's not something static). But that can be done later ofc.